### PR TITLE
style fixes for Timberland libnvme

### PR DESCRIPTION
**Pull request by mwilck**
These patches address https://github.com/timberland-sig/libnvme/pull/4#issuecomment-1482571536

### CODE REVIEW

**Comment by tbzatek** [comment]
Well that was mostly what I had ready locally for my second batch of changes, just haven't pushed that yet as I had some other stuff in works. Are you planning more changes here?

**Comment by mwilck** [comment]
If this is just duplicate work, I have no issue with just closing it.

**Comment by tbzatek** [comment]
I don't know, it's kinda sad that we both worked on the same code while I mentioned yesterday on the call that I have second batch of changes coming that would address the rest of the issues. I've pushed my code https://github.com/timberland-sig/libnvme/tree/timberlandmaster_final_v13-fixes-2 in the meantime but it's still WIP and needs to undergo verification on real deployment here...

**Comment by mwilck** [comment]
I rebased this on top of your [timberlandmaster_final_v13-fixes-2](https://github.com/timberland-sig/libnvme/tree/timberlandmaster_final_v13-fixes-2) branch now.

Turns out that only part of our work was duplicate. I hope this is ok for you.

File: src/nvme/nbft.h [hunk 1]
```diff
@@ -1124,7 +1124,6 @@ struct nbft_info_discovery {
  */
 struct nbft_info_security {
 	int index;
-	/* TODO add fields */
```
Comment (tbzatek) [inline src/nvme/nbft.h:1046]: I still plan to add those fields ;-)

**Review by tbzatek — APPROVED** [review]
Good point about the packed attrs, actually nice trick with the compiler warning switch, I was not aware of it!

Tested locally against the NBFT table dumps and everything seems to be in order on a first sight, not tested on a real deployment though.

So looks good from my point of view.

**Comment by tbzatek** [comment]
A side note for the packed attrs: I haven't actually checked if all the structures required the packed attr, only a couple of those were showing different sizeof() values during my initial testing. The current changes appear to be working fine, would be nice to test it on 32-bit arch (hmm, would anyone ever use NBFT on 32-bit...?), I hope to have it covered with my upcoming (unit) test case.

The edk2 definitions are all packed, it's all about a specific binary format that needs to fit after all:
https://github.com/timberland-sig/edk2/blob/a32ad775d533fa945fdc4515d92bbb21c8b90fb9/MdePkg/Include/IndustryStandard/NvmeOfBootFirmwareTable.h#L41

We still can use pahole to optimize the _convenient_ / `nbft_info_` structs layout as they're not read from binary data, that's another (perhaps unnecessary) story anyway.

**Comment by johnmeneghini** [comment]
So what's the status of this pull request?  Do we want to merge this?

**Comment by johnmeneghini** [comment]
I have incorporated these changes into the `timberlandmaster_final_v13-fixes` branch.  Closing this merge request.

**Pull request closed without merging**